### PR TITLE
added dark and light theme switch capability

### PR DESCRIPTION
**Pull request by dcook3**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,35 +1,26 @@
...
-      <div className="container">
-        <footer className="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
-          <div className="col-md-4 d-flex align-items-center">
-            <Link className='text-decoration-none' to="/">Retailer Site Name</Link>
-            <span className="text-muted ms-1">| © 2022 Dylan Cook</span>
-          </div>
-        </footer>
-      </div>
-    </>
+    <ThemeContext.Provider value={{theme: theme}}>
```
Comment (bowen9284) [inline src/App.js:19]: Don't forget we have an opportunity to refactor our Context as seen here https://neit.instructure.com/courses/53221/pages/live-coding-ii-refactoring-opportunities?module_item_id=2248881 and in the video https://neit.instructure.com/courses/53221/pages/video-live-coding-ii-refactoring-opportunities?module_item_id=2268220

File: src/App.js [hunk 2]
```diff
@@ -1,35 +1,26 @@
...
+import {Outlet} from 'react-router-dom';
+import Main from './components/Main';
+import { ThemeContext, themes } from './context/themeContext';
+import { useState, useContext } from 'react';
+import Navbar from './components/Navbar'
+import Button from './components/Button'
 
-function App() {
+function Providers() {
+  const [theme, setTheme] = useState(themes.dark);
```
Comment (bowen9284) [inline src/App.js:11]: The entire background does not change with the theme, half of the application remains white at all times. The fix required a little digging on your end but this is how you'd do it.
```
document.body.style.background = theme.background;
```
If you end up refactoring based on the links I posted, then you'd want this piece of code within the useEffect of your provider.

File: src/components/Navbar.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    return (
+        <nav className="navbar navbar-expand-md px-2" style={{backgroundColor: theme.foreground}}>
+            <Link className='navbar-brand' to="/">Site Brand</Link>
+            <div className='container-fluid'>
+                <div className="navbar-nav">
+                    <Link className='nav-link' to="/Products">Products</Link>
+                    <Link className='nav-link' to="/About">About</Link>
+                </div>
+                <button className="btn btn-primary" onClick={props.toggleTheme}><img src={themeIcon}></img></button>
```
Comment (bowen9284) [inline src/components/Navbar.js:17]: It would be cool if the button changed color too 😄

**Review by bowen9284 — APPROVED** [review]
Looks good Dylan. It would be better if you applied the refactoring piece as shared on Canvas. I'm taking half of the points off of the rubric item `All backgrounds change color when toggling a theme` since half of the background is not changing with any theme. 

88/100

File: src/App.js (hunk 2 above)
Comment (dcook3) [reply src/App.js:11]: Life saver! This was stumping me and I kept trying to change the height styles to fix this but your code just does it without having to worry about height. Thanks!

File: src/components/Navbar.js (hunk 3 above)
Comment (dcook3) [reply src/components/Navbar.js:17]: Okay but only on the navbar! It would be a real pain to change all of the buttons in the product components.

**Pull request merged**